ui/lib/config-ui/addon/routes/messages/index.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+import { hash } from 'rsvp';
+
+export default class MessagesRoute extends Route {
+  @service store;
+
+  queryParams = {
+    page: {
+      refreshModel: true,
+    },
+    authenticated: {
+      refreshModel: true,
+    },
+    pageFilter: {
+      refreshModel: true,
+    },
+  };
+
+  async model(params) {
+    try {
+      const { authenticated, page, pageFilter } = params;
+      const filter = pageFilter
+        ? (dataset) => dataset.filter((item) => item?.title.toLowerCase().includes(pageFilter.toLowerCase()))
+        : null;
+      const messages = await this.store.lazyPaginatedQuery('config-ui/message', {
+        authenticated,
+        pageFilter: filter,
+        responsePath: 'data.keys',
+        page: page || 1,
+        size: 10,
+      });
+      return hash({
+        pageFilter,
+        messages,
+      });
+    } catch (e) {
+      if (e.httpStatus === 404) {
+        return [];
+      }
+
+      throw e;
+    }
+  }
```

Comment:
Currently, `hash` is not necessary since you are awaiting the messages promise. Here's an alternative that would leverage `hash` and simplify this a bit:

```suggestion
  model(params) {
    const { authenticated, page, pageFilter } = params;
    const filter = pageFilter
      ? (dataset) => dataset.filter((item) => item?.title.toLowerCase().includes(pageFilter.toLowerCase()))
        : null;
    const messages = this.store.lazyPaginatedQuery('config-ui/message', {
        authenticated,
        pageFilter: filter,
        responsePath: 'data.keys',
        page: page || 1,
        size: 10,
      }).catch((e) => {
        if (e.httpStatus === 404) {
          return [];
        }
        throw e;
      });
      return hash({
        pageFilter,
        messages,
      });
  }
```